src/prefect/server/models/workers.py

```diff
@@ -135,6 +135,31 @@ async def read_work_pools(
     return result.scalars().unique().all()
 
 
+@inject_db
+async def count_work_pools(
+    db: PrefectDBInterface,
+    session: AsyncSession,
+    work_pool_filter: schemas.filters.WorkPoolFilter = None,
+) -> int:
+    """
+    Read worker configs.
+
+    Args:
+        session: A database session
+        work_pool_filter: filter criteria to apply to the count
+    Returns:
+        int: the count of work pools matching the criteria
+    """
+
+    query = select(sa.func.count(sa.text("*"))).select_from(db.WorkPool)
```

Comment:
I _think_ you can skip the `*`:

```suggestion
    query = select(sa.func.count()).select_from(db.WorkPool)
```